modules/ismp/pallets/token-gateway/src/benchmarking.rs

```diff
@@ -0,0 +1,191 @@
+
+        #[extrinsic_call]
+        _(RawOrigin::Signed(account), asset);
+
+        Ok(())
+    }
+
+    #[benchmark]
+    fn teleport() -> Result<(), BenchmarkError> {
+        let account: T::AccountId = whitelisted_caller();
+
+        let asset_details = GatewayAssetRegistration {
+            name: BoundedVec::try_from(b"Spectre".to_vec()).unwrap(),
+            symbol: BoundedVec::try_from(b"SPC".to_vec()).unwrap(),
+            chains: vec![StateMachine::Evm(100)],
+            minimum_balance: None,
+        };
+        let asset = create_dummy_asset::<T>(asset_details);
+
+        Pallet::<T>::create_erc6160_asset(RawOrigin::Signed(account.clone()).into(), asset)?;
+
+        let dummy_teleport_params = dummy_teleport_asset::<T>();
+
+        // Set balances
+        let ed = <T as pallet_balances::Config>::ExistentialDeposit::get();
+
+        // Adjust total issuance
+        pallet_balances::Pallet::<T>::force_adjust_total_issuance(
+            RawOrigin::Root.into(),
+            AdjustmentDirection::Increase,
+            ed * 1000,
+        )?;
+
+        let acc = <<T as frame_system::Config>::Lookup as StaticLookup>::unlookup(account.clone());
+
+        pallet_balances::Pallet::<T>::force_set_balance(
+            RawOrigin::Root.into(),
+            acc.clone(),
+            ed * 100u128,
+        )?;
+
+        #[extrinsic_call]
+        teleport(RawOrigin::Signed(account), dummy_teleport_params);
+        Ok(())
+    }
+
+    #[benchmark]
+    fn set_token_gateway_addresses(x: Linear<5, 100>) -> Result<(), BenchmarkError> {
+        let mut addresses = BTreeMap::new();
+        for i in 0..x {
+            let addr = i.to_string().as_bytes().to_vec();
+            addresses.insert(StateMachine::Evm(100), addr);
+        }
+
+        #[extrinsic_call]
+        _(RawOrigin::Root, addresses);
+        Ok(())
+    }
+
+    #[benchmark]
+    fn update_erc6160_asset() -> Result<(), BenchmarkError> {
+        let acc_origin: T::AccountId = whitelisted_caller();
+
+        let asset_details = GatewayAssetRegistration {
+            name: BoundedVec::try_from(b"Spectre".to_vec()).unwrap(),
+            symbol: BoundedVec::try_from(b"SPC".to_vec()).unwrap(),
+            chains: vec![StateMachine::Evm(100)],
+            minimum_balance: None,
+        };
+        let asset = create_dummy_asset::<T>(asset_details);
+        let asset_id: H256 = sp_io::hashing::keccak_256(asset.reg.symbol.as_ref()).into();
+
+        // set balances
+
+        let acc_o =
+            <<T as frame_system::Config>::Lookup as StaticLookup>::unlookup(acc_origin.clone());
+        let ed = <T as pallet_balances::Config>::ExistentialDeposit::get();
+        pallet_balances::Pallet::<T>::force_set_balance(
+            RawOrigin::Root.into(),
+            acc_o.clone(),
+            ed * 100u128,
+        )?;
+
+        // set asset balance
+        pallet_assets::Pallet::<T>::create(
```

Comment:
Use the `fungibles::Mutate` trait to also do this instead of pulling in `pallet-assets`